libtelemetry/src/main/java/com/mapbox/android/telemetry/CertificatePinnerFactory.java

```diff
@@ -36,11 +39,40 @@ CertificatePinner provideCertificatePinnerFor(Environment environment) {
     return CERTIFICATES_PINS.get(environment);
   }
 
-  private void addCertificatesPins(Map<String, List<String>> pins, CertificatePinner.Builder builder) {
+  private void addCertificatesPins(Map<String, List<String>> pins, CertificatePinner.Builder builder,
+                                   Environment environment) {
+    pins = removeBlacklistedPins(pins, environment);
+
     for (Map.Entry<String, List<String>> entry : pins.entrySet()) {
       for (String pin : entry.getValue()) {
         builder.add(entry.getKey(), pin);
       }
     }
   }
+
+  private Map<String, List<String>> removeBlacklistedPins(Map<String, List<String>> pins, Environment environment) {
+    String key = COM_EVENTS;
+
+    if (environment == Environment.CHINA) {
+      key = CHINA_EVENTS;
+    }
+
+    List<String> hashList = pins.get(key);
+    CertificateBlacklist certificateBlacklist = new CertificateBlacklist(MapboxTelemetry.applicationContext,
```

Comment:
We're coupling the retrieve blacklist functionality to `CertificatePinnerFactory`. What about passing the blacklist somehow from `MapboxTelemetry` all the way down to `CertificatePinnerFactory`? Same idea as pins current implementation:

`MapboxTelemetry#createTelemetryClient` > `TelemetryClientFactory#obtainTelemetryClient` > `TelemetryClient` > `TelemetryClient#sendBatch` - `setting.getClient()` > `TelemetryClientSettings#configureHttpClient` > `CertificatePinnerFactory#provideCertificatePinnerFor`

IMO `CertificatePinnerFactory` shouldn't be responsible of retrieving the blacklist. Does that make sense? `

Reply:
What is the tangible benefit of this much more complicated routing? What are the cons of the current implementation?

Reply:
The main goal of the change is being able to test the new certificate blacklist functionality. Right now, we're not able to inject a `CertificateBlacklist` instance because we're creating a new instance of it within `CertificatePinnerFactory#removeBlacklistedPins`. We're calling `retrieveBlackList` which ends up retrieving the `File`, reading it and returning a `List` of its content which is not responsibility of `CertificatePinnerFactory`. Also, as commented in https://github.com/mapbox/mapbox-events-android/pull/151#discussion_r202375564 in order to access to the `accessToken` (needed for creating a `CertificateBlacklist`) we had to make it "globally" accessible in `MapboxTelemetry` which shouldn't be necessary.